pkg/installer/install

```diff
@@ -477,7 +478,11 @@ if [ "$INSTALL_ZFS" = true ]; then
       done
   fi
 else
-  P3="$(find_part P3 "$INSTALL_DEV")"
+  if [ "$DISK_WITH_P3" != "" ]; then
+      P3="$(find_part P3 "$DISK_WITH_P3")"
+  else
+      P3="$(find_part P3 "$INSTALL_DEV")"
+  fi
   [ -z "$P3" ] && bail "Installation failed. Cannot found P3. Entering shell..."
```

Comment:
I am trying to understand the current and future logic.

Current logic is contained in `find_part P3 $INSTALL_DEV`, which means, "find the P3 partition on my main install disk." So if I am installing to sda, it will find P3 partition on sda.

Proposed logic is:

1. If `INSTALL_DEV != INSTALL_PERSIST`, i.e. if the 2 are separate devices, then set `DISK_WITH_P3 = $INSTALL_PERSIST` - 
2. If `DISK_WITH_P3` is not blank, then find P3 partition on `DISK_WITH_P3`, else find it on `INSTALL_DEV`

So it relies on `"$INSTALL_DEV" != "$INSTALL_PERSIST"` to determine if they are different, sets `DISK_WITH_P3` as a flag, and if the flag is set, then look on that disk. Is that it?

Makes sense, just a bit hard to read.

Would it be easier to just change line 480 to:

```sh
  P3="$(find_part P3 "$INSTALL_PERSIST")"
```

By definition, `INSTALL_PERSIST` either is equal to `INSTALL_DEV`, in which case (in PR code) `DISK_WITH_P3` is blank and therefore it is equal to the currentl or it is different, in which case we just want to look at `INSTALL_DEV`.

Reply:
You are right @deitch , I was just trying to follow the same logic applied in the whole script (by having the DISK_WITH_P3 variable), but it doesn't make sense to turn things more complex....